service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -0,0 +1,768 @@
+#!/bin/bash
+#
+# Name: startup.sh
+#
+# Description
+#   Initialization script to setup Terra configurations in a Dataproc cluster manager node
+#
+# Execution details
+#   This script is executed as root on all Dataproc nodes.
+#
+#   Note that the guest attribute is set to DONE whether the script runs successfully or not.
+#
+# How to test changes to this file:
+# Currently, the only way is to use swagger to create a new cluster with the startup script gs url passed into the 'startup-script-uri' metadata field.
+#   TODO: Pending CLI support to create a cluster with a custom startup script via CLI
+#
+# Integration Tests
+#   Please also make sure integration test `PrivateControlledDataprocClusterStartup` passes. Refer to
+#   https://github.com/DataBiosphere/terra-workspace-manager/tree/main/integration#Run-nightly-only-test-suite-locally
+#   for instruction on how to run the test.
+
+# Only run on the dataproc manager node. Exit silently if otherwise.
+readonly ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+if [[ "${ROLE}" != 'Master' ]]; then exit 0; fi
+
+# Only run on first startup
+if [[ -f /etc/startup_was_launched ]]; then exit 0; fi
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+# The linux user that JupyterLab will be running as. It's important to do some parts of setup in the
+# user space, such as setting Terra CLI settings which are persisted in the user's $HOME.
+# This post startup script is not run by the same user.
+readonly DATAPROC_USER="dataproc"
+
+# Create an alias for cases when we need to run a shell command as the dataproc user.
+# Note that we deliberately use "bash -l" instead of "sh" in order to get bash (instead of dash)
+# and to pick up changes to the .bashrc.
+#
+# This is intentionally not a Bash function, as that can suppress error propagation.
+# This is intentionally not a Bash alias as they are not supported in shell scripts.
+readonly RUN_AS_DATAPROC_USER="sudo -u ${DATAPROC_USER} bash -l -c"
```

Comment:
I think the need to sweep the code for RUN_AS_JUPYTER_USER -> RUN_AS_DATAPROC_USER bothers me most from a maintenance standpoint. It makes it harder to diff and keep the two scripts in sync. I'm trying to come up with the right nomenclature such that this script and the Vertex AI post-startup script can be coded consistently (but struggling a bit). A few thoughts:

- RUN_AS_NON_ROOT_USER
- RUN_AS_LOCAL_USER
- RUN_AS_LOGIN_USER
- RUN_AS_TARGET_USER

The idea would be to use the above and make the code consistent across the two scripts.
I think I like `RUN_AS_LOGIN_USER` the best of that group. It's not perfect, but feels like with a good comment and a little bit of time, it would be good enough.